### ISSUE TITLE
Tab Stash-specific context menu in sidebar

### ISSUE THREAD

**Issue by mforkel** [josh-berry/tab-stash#253]
### Problem Statement

There seems to be no specific context menu for collapsed items in the TS sidebar. Some of the commands in the current context either have no effect (e.g. View Page Source) or are not really applicable (e.g. Tab Stash > Copy This Tab to Stash). FF-native sidebars seem to have their own context menus, e.g. the bookmarks sidebar.

### Preferred Solution(s)

Replace or adapt the current sidebar context menu for collapsed items. Commands in the menu might include additional commands like 'Open Tabs in New Window' as well as some of the commands already provided by the pop-up icon bar, e.g. 'Delete this group'.

### Additional Context

The commands provided the sidebar context menu might be useful in the context menus of other views as well, e.g. in the Stashed Tabs tab.

<hr>
<em>Readers: If you are also interested in seeing this feature be developed, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the feature request, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Thanks for the suggestion! Unfortunately, Firefox gives extensions only limited ability to control what happens in the built-in context menu—e.g. there's no way for the context menu to know if it was opened on a link, a section header, etc.

Another approach might be to replace the context menu entirely with a custom one, but this is something I've avoided doing so far since users might want to use items from other extensions in the Tab Stash UI (one example might be generating a QR code from a saved tab—something Tab Stash doesn't do natively, but can be done by other extensions that have their own context-menu items).

I'll keep this in mind for the future, though, in case anything changes.

**Comment by mforkel** [comment]
Thanks for the explanation! I was not aware of those limitations. Does an extension have the ability to tell whether a context menu is opened in a regular page or a sidebar?

**Comment by josh-berry** [comment]
> Thanks for the explanation! I was not aware of those limitations. Does an extension have the ability to tell whether a context menu is opened in a regular page or a sidebar?

No, not until a menu item is actually selected. Basically the way it works is, you define the structure you want for the menu ahead of time, and the browser will display that structure when requested by the user. Then the extension gets notified when one of its menu items is chosen, and info about the currently-selected tab (or sidebar) etc. is sent along with that notification.
